### DISCUSSION TITLE
Drop FS persistence

### DISCUSSION THREAD

**Discussion by paulmelnikow** [badges/shields#5665] [Ideas] [answered]
We switched over to cloud Redis for token persistence in #3332 over a year ago, and haven't looked back. Now that the servers are on Heroku we can't even go back to FsTokenPersistence in production since there's no persistent filesystem.

The only reason to maintain FsTokenPersistence would be to support self-hosting users. However, I really doubt any self-hosting users are using this ability to store tokens in a JSON file on the server. I highly doubt any of them would require more than one GitHub token.

(If they did, the way I think we ought to support it is to allow multiple static GitHub tokens via the ordinary config, rather than requiring them to check in an extra file with tokens. This is mentioned in the top comment of #2733 and I think we can track it there. Again, I don't think any self-hosting users are using it now.)

Removing this will help pick away at #2733 and continue to reduce both the amount of code and the number of different approaches to GitHub auth we support.

Do folks agree with removing this?

**Comment by calebcartwright** [comment]
> Do folks agree with removing this?

Yup, agreed :+1:

**Comment by paulmelnikow** [comment, marked as answer]
This was dealt with in #5671.
